core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1858,13 +1865,40 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     else dataAccess.workspaceQuery.getUserSharePermissions(RawlsUserSubjectId(userInfo.userSubjectId), workspaceContext)
   }
 
-  private def withEntity(workspaceContext: SlickWorkspaceContext, entityType: String, entityName: String, dataAccess: DataAccess)(op: (Entity) => ReadWriteAction[PerRequestMessage]): ReadWriteAction[PerRequestMessage] = {
+  private def withEntity[T](workspaceContext: SlickWorkspaceContext, entityType: String, entityName: String, dataAccess: DataAccess)(op: (Entity) => ReadWriteAction[T]): ReadWriteAction[T] = {
     dataAccess.entityQuery.get(workspaceContext, entityType, entityName) flatMap {
       case None => DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.NotFound, s"${entityType} ${entityName} does not exist in ${workspaceContext.workspace.toWorkspaceName}")))
       case Some(entity) => op(entity)
     }
   }
 
+  private def withAllEntities[T](workspaceContext: SlickWorkspaceContext, dataAccess: DataAccess, entities: Seq[AttributeEntityReference])(op: (Seq[Entity]) => ReadWriteAction[T]): ReadWriteAction[T] = {
+    val entityActions: Seq[ReadAction[Try[Entity]]] = entities map { e =>
+      dataAccess.entityQuery.get(workspaceContext, e.entityType, e.entityName) map {
+        case None => Failure(new RawlsException(s"${e.entityType} ${e.entityName} does not exist in ${workspaceContext.workspace.toWorkspaceName}"))
+        case Some(entity) => Success(entity)
+      }
+    }
+
+    DBIO.sequence(entityActions) flatMap { entityTries =>
+      val failures = entityTries.collect { case Failure(y) => y.getMessage }
+      if (failures.size == entityTries.size) {
```

Comment:
i think you can reasonably merge this block with the one below, and return the full list in the case where none of them can be found